crates/forge-runner/src/cairo_runner/casm_run.rs

```diff
@@ -0,0 +1,105 @@
+use cairo_felt::Felt252;
+use cairo_lang_casm::instructions::Instruction;
+use cairo_lang_runner::casm_run::RunFunctionContext;
+use cairo_vm::hint_processor::hint_processor_definition::HintProcessor;
+use cairo_vm::serde::deserialize_program::{BuiltinName, HintParams, ReferenceManager};
+use cairo_vm::types::program::Program;
+use cairo_vm::types::relocatable::MaybeRelocatable;
+use cairo_vm::vm::errors::cairo_run_errors::CairoRunError;
+use cairo_vm::vm::runners::cairo_runner::CairoRunner;
+use cairo_vm::vm::vm_core::VirtualMachine;
+use std::collections::HashMap;
+
+#[allow(dead_code)]
+type RunFunctionRes = (Vec<Option<Felt252>>, usize);
+
+pub fn run_function_with_runner(
+    vm: &mut VirtualMachine,
+    data_len: usize,
+    additional_initialization: fn(
+        context: RunFunctionContext<'_>,
+    ) -> Result<(), Box<CairoRunError>>,
+    hint_processor: &mut dyn HintProcessor,
+    runner: &mut CairoRunner,
+) -> Result<(), Box<CairoRunError>> {
+    let end = runner.initialize(vm).map_err(CairoRunError::from)?;
+
+    additional_initialization(RunFunctionContext { vm, data_len })?;
+
+    runner
+        .run_until_pc(end, vm, hint_processor)
+        .map_err(CairoRunError::from)?;
+    runner
+        .end_run(true, false, vm, hint_processor)
+        .map_err(CairoRunError::from)?;
+    runner.relocate(vm, true).map_err(CairoRunError::from)?;
+    Ok(())
+}
+
+pub fn build_runner(
```

Comment:
```suggestion
pub fn build_cairo_runner(
```
Also I would split it into build program and build runner. as this one builds program more than runner.